### PR TITLE
Test multiple generator kwargs

### PR DESCRIPTION
**Pull request by aaron-sandoval**
Addresses #11 
In `test_latticemaze.py`, all instances of testing over everything in `GENERATORS_MAP` have been replaced by iterating over `DEFAULT_GENERATORS`. Special exceptions for generation methods that would stochastically fail are removed.

### CODE REVIEW

**Comment by aaron-sandoval** [comment]
Environment setup issues currently have this branch with a very different poetry.lock file. Still need to get tests to work with the original poetry.lock from the main branch before merging.

File: ORIG_poetry.lock
Comment (mivanit) [inline ORIG_poetry.lock:1]: This file should just be deleted -- we updated the dependencies, so the the new `poetry.lock` is what we should care about testing with. If we ever need the old one for diagnosing an error, we can always find it in the commit history

**Review by mivanit — CHANGES_REQUESTED** [review]
No need for the old `ORIG_poetry.lock` file, and the formatter needs to be run. Other than that, LGTM!

**Comment by mivanit** [comment]
looks like this might be failing due to black being bumped to version 24 (formatting checks don't respect our `pyproject.toml` deps unfortunately). maybe roll that dep update into this PR? Don't squash and merge if this PR ends up including that format run, just do a regular merge

**Comment by mivanit** [comment]
looks great! Thanks for taking care of this

**Pull request merged**